pf9/support/generate_bundle.py

```diff
@@ -0,0 +1,199 @@
+from pf9.modules.express import Get
+from pf9.modules.util import Utils, Logger, Pf9ExpVersion
+
+class Log_Bundle:
+
+    def __init__(self,error_msg="none"):
+        self.error_msg = error_msg
+
+    def get_uuid_from_resmgr(self,du_url,host_ip,headers):
+    #Based on the host_ip this function would pull the host UUID from resmgr.
+        try:
+            api_endpoint = "resmgr/v1/hosts"
+            pf9_response = requests.get("{}/{}".format(du_url,api_endpoint),
+                                        headers=headers)
+            if pf9_response.status_code != 200:
+                return None
+            # parse resmgr response
+            json_response = json.loads(pf9_response.text)
+        except Exception as except_err:
+            logger.exception(except_err)
+            return None
+        # sequentially search resmgr response for the ip of the host, if the ip is found return host uuid
+        for host in json_response:
+            try:
+                for iface_name, iface_ip in host['extensions']['interfaces']['data']['iface_ip'].items():
+                    if iface_ip == host_ip:
+                        logger.info("Node host_id: {}".format(host['id']))
+                        return host['id']
+            except Exception as e:
+                logger.exception(e)
+        return None
+
+    def check_host_status(self,ctx,ips,user,password):
+        """ Check the status of the host in Resource manager , obtaing the UUID and verify the role and responding status """
+        Get(ctx).active_config()
+        du_url = ctx.params['du_url']
+        token = ctx.params['token']
+        host_uuid = {}
+        headers = {'Content-Type': 'application/json', 'X-Auth-Token': token}
+        for host in ips:
+            #If the host provided is local host
+            # get ip of the localhost and get the hostname
+            if host in ['127.0.0.1','localhost']:
+                hostname=socket.gethostname()
+                ip=socket.gethostbyname(hostname)
+            # get host uuid from resmgr using the ip of the localhost
+                host_uuid[ip]=self.get_uuid_from_resmgr(du_url,ip,headers)
+
+                if host_uuid[ip]:
+                # if a valid uuid is found , hostagent is configured and we can pull the support bundle from the host
+                    resmgr_endpoint = '{}/resmgr/v1/hosts/{}'.format(ctx.params['du_url'],host_uuid[ip])
+                    resmgr_get_hosts = requests.get(resmgr_endpoint,verify=False,headers=headers)
+                    data = resmgr_get_hosts.json()
+                    roles = (data["roles"])
+                    if len(roles) >= 1 :
+                        if roles[0] == "pf9-kube" and data["info"]["responding"] == "true":  # Responding key will only be there is pf9-kube is installed else there will not be any responding key in response
+                            pass
+                        else:
+                            time.sleep(2)
```

Comment:
Why sleep here? And also few lines below.

Reply:
Removed the sleep from the code. It was added earlier for the user prompt and progress bar which we have now removed.